src/components/inputs/Search/SearchBar/SearchBar.tsx

```diff
@@ -17,30 +19,44 @@ function SearchBar(props: SearchBarProps) {
   function reset() {
     setSearchSubmitted(false);
     setSubmittedName('');
-    setIsValid(false);
+    setIsValid(true);
     setIsDefault(true);
     setSearchResult('');
     return;
   }
 
   function onHandleChange(e: React.ChangeEvent<HTMLInputElement>) {
-    setSearchBarText(e.target.value.trim());
+    if (
+      ARNS_NAME_REGEX.test(e.target.value) ||
+      e.target.value.length < 1 ||
+      (e.target.value[e.target.value.length - 1] == '-' &&
+        e.target.value.length > 1)
+    ) {
+      setSearchBarText(e.target.value);
+      setIsValid(true);
+    } else {
+      setIsValid(false);
+    }
     if (e.target.value === '') {
       reset();
     }
```

Comment:
put this first and return after, then you can remove the <1 parameters I believe

Reply:
no still need that for if a user wants to completely delete the string. The second <1 is to allow dashes to be input on the end, which fails the regex, but is caught in the onSubmit function instead.